### PR TITLE
Removing code duplication in addon logic

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: pkg/acsengine/artifacts.go [hunk 1]
```diff
@@ -55,27 +55,27 @@ func kubernetesAddonSettingsInit(profile *api.Properties) []kubernetesFeatureSet
...
 			"kubernetesmasteraddons-aci-connector-deployment.yaml",
 			"aci-connector-deployment.yaml",
-			profile.OrchestratorProfile.KubernetesConfig.IsACIConnectorEnabled(),
+			profile.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(DefaultACIConnectorAddonName, api.DefaultACIConnectorAddonEnabled),
 		},
 		{
 			"kubernetesmasteraddons-cluster-autoscaler-deployment.yaml",
 			"cluster-autoscaler-deployment.yaml",
-			profile.OrchestratorProfile.KubernetesConfig.IsClusterAutoscalerEnabled(),
+			profile.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(DefaultClusterAutoscalerAddonName, api.DefaultACIConnectorAddonEnabled),
```
[the remark this answers is not in the document]
Comment (tariq1890) [reply pkg/acsengine/artifacts.go]: :'( Yes. very much. will Fix

File: pkg/acsengine/artifacts.go [hunk 2]
```diff
@@ -130,17 +130,17 @@ func kubernetesAddonSettingsInit(profile *api.Properties) []kubernetesFeatureSet
...
 			"kubernetesmasteraddons-smb-flexvolume-installer.yaml",
 			"smb-flexvolume-installer.yaml",
-			profile.OrchestratorProfile.KubernetesConfig.IsSMBFlexVolumeEnabled(),
+			profile.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(DefaultSMBFlexVolumeAddonName, api.DefaultSMBFlexVolumeAddonEnabled),
 		},
 		{
 			"kubernetesmasteraddons-keyvault-flexvolume-installer.yaml",
 			"keyvault-flexvolume-installer.yaml",
-			profile.OrchestratorProfile.KubernetesConfig.IsKeyVaultFlexVolumeEnabled(),
+			profile.OrchestratorProfile.KubernetesConfig.IsAddonEnabled(DefaultKeyVaultFlexVolumeAddonName, api.DefaultSMBFlexVolumeAddonEnabled),
```
[the remark this answers is not in the document]
Comment (tariq1890) [reply pkg/acsengine/artifacts.go]: :'(

**Comment by jackfrancis** [comment]
Thanks for doing this @tariq1890! ❤️

**Pull request merged**